pulpcore/pulpcore/app/views/content.py

```diff
@@ -285,18 +288,56 @@ def _dispatch(self, request, path):
             django.http.HttpResponseForbidden: on forbidden.
             django.http.HttpResponseRedirect: on redirect to the streamer.
         """
+        methods = {
+            'django': self._django,
+            'apache': self._apache,
+            'nginx': self._nginx,
+        }
+
         server = settings.CONTENT['WEB_SERVER']
 
         try:
-            responder = self.RESPONDER[server]
+            responder = methods[server]
         except KeyError:
             raise ValueError(_('Web server "{t}" not supported.').format(t=server))
         else:
             disposition = os.path.basename(request.path)
-            response = responder(self, path)
+            response = responder(path)
             response['Content-Disposition'] = 'attachment; filename={n}'.format(n=disposition)
             return response
 
+    def _permit(self, request, distribution):
+        """
+        Permit the request.
+
+        Authorization is delegated to the optional content-guard associated with the distribution.
+
+        Args:
+            request (django.http.HttpRequest): A request for a published file.
+            distribution (Distribution): The matched distribution.
+
+        Raises:
+            PermissionError: When not permitted.
```

Comment:
Showing the full class path to `PermissionError` here would be helpful I think since its not a standard Exception.